pkg/operator/startupmonitorreadiness/readiness_checks.go

```diff
@@ -275,11 +276,20 @@ func goodHealthzEtcdEndpoint(client *http.Client, rawURL string) func(context.Co
 func doHTTPCheckAndTransform(ctx context.Context, client *http.Client, rawURL string, checkName string, httpCheckFn func(ctx context.Context, client *http.Client, rawURL string) (int, string, error)) (bool, string, string) {
 	statusCode, response, err := httpCheckFn(ctx, client, rawURL)
 	if err != nil {
+		if utilnet.IsConnectionRefused(err) {
+			return false, "ConnectionRefused", fmt.Sprintf("kube-apiserver is not reachable: %v", err)
+		}
+		if utilnet.IsConnectionReset(err) || utilnet.IsProbableEOF(err) {
+			return false, "ConnectionError", fmt.Sprintf("failed sending request to kube-apiserver: %v", err)
```

Comment:
"waiting for kube-apiserver static pod %s to answer requests: %v"

Reply:
mhm, we are not waiting, we failed possibly due to some network errror

Reply:
the whole startup-monitor is waiting. I would formulate these in that way. Maybe this one is different as it is a random error. But for connection refused we are waiting for the process and port.